### PR TITLE
EasyConfig for CMake 3.6.1 with no particular toolchain

### PR DESCRIPTION
**Pull request by valtandor**

### CODE REVIEW

File: easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+source_urls = ['http://www.cmake.org/files/v%(version_major_minor)s']
+sources = [SOURCELOWER_TAR_GZ]
+
+configopts = '-- -DCMAKE_USE_OPENSSL=1'
+
+# Don't depend on anything, since we will use this to build other software
+# which might want different versions of the same dependency, eg: ncurses
+dependencies = []
```
Comment (boegel) [inline easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb]: I understand the motivation behind this, but then we should list ncurses (probably `ncurses-devel` on RHEL-based systems) as an OS dependency?

File: easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel')]
+
+sanity_check_paths = {
+    'files': ["bin/%s" % x for x in ['cmake', 'cpack', 'ctest']],
```
Comment (boegel) [inline easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb]: please also list `ccmake` here, cfr. #4196

File: easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+source_urls = ['http://www.cmake.org/files/v%(version_major_minor)s']
+sources = [SOURCELOWER_TAR_GZ]
+
+configopts = '-- -DCMAKE_USE_OPENSSL=1'
+
+# Don't depend on anything, since we will use this to build other software
+# which might want different versions of the same dependency, eg: ncurses
+dependencies = []
+
+osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel', 'ncurses-devel')]
```
Comment (boegel) [inline easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb]: This should be:

```python
osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel'), 'ncurses-devel']
```

The way you have it now, EasyBuild will stop checking as soon as one of the packages for OpenSSL is available; only one of the entries in `(...)` has to be available.

File: easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+sanity_check_paths = {
+    'files': ["bin/%s" % x for x in ['cmake', 'ccmake', 'cpack', 'ctest']],
```
Comment (boegel) [inline easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb]: nitpicking: `ccmake` should come before `cmake` (alphabetically sorted)

File: easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb (hunk 1 above)
Comment (valtandor) [inline easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb]: @boegel Do we need these three lines (17 to 19) for a purpose? I can take them out if they're not needed for standardisation and/or information.

Comment (boegel) [inline easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb]: @valtandor I guess specifying `dependencies = []` doesn't make much sense

Maybe you should replace this block with a comment above `osdependencies` like this:

```python
# use OS dependencies to ensure that CMake can be used to build software that depends on them
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2423.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/55690bcabe04f63d1c46d54d2ad9709c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2014.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/19a75513090c3c38ccb6938b15d61eaf for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2057.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/cab6e18834999cec9fbc59a2b9516d6b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2495.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/012a20bdf6806df96f690fe00562ce82 for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @valtandor!

**Pull request merged**